src/Microsoft.Framework.ConfigurationModel.Interfaces/IConfigurationSourceRoot.cs

```diff
@@ -7,6 +7,8 @@ namespace Microsoft.Framework.ConfigurationModel
 {
     public interface IConfigurationSourceRoot : IConfiguration
     {
+        string BasePath { get; }
```

Comment:
- should this instead be on `IConfiguration` and in the `Configuration` constructor?  would avoid odd cases where configuration sources in the same configuration have different base paths.  would also avoid callers repeatedly passing the same value for multiple configuration sources.
- (not sure of the docs rules for this repo but) shouldn't this new property be documented?